test/ec_internal_test.c

```diff
@@ -259,6 +259,39 @@ static int underflow_test(void)
 }
 #endif
 
+/*
+ * Tests behavior of the EC_KEY_set_private_key
+ */
+static int set_private_key(void)
+{
+    EC_KEY *key = NULL, *aux_key = NULL;
+    int testresult = 0;
+
+    key = EC_KEY_new_by_curve_name(NID_secp224r1);
+    aux_key = EC_KEY_new_by_curve_name(NID_secp224r1);
+    if (!TEST_ptr(key)
+        || !TEST_ptr(aux_key)
+        || !TEST_int_eq(EC_KEY_generate_key(key), 1)
+        || !TEST_int_eq(EC_KEY_generate_key(aux_key), 1))
+        goto err;
+
+    /* Test setting a valid private key */
+    if (!TEST_int_eq(EC_KEY_set_private_key(key, aux_key->priv_key), 1))
```

Comment:
I think I would prefer `TEST_int_eq(x, 1)` here, as we want to explicitly test for `1` (rather than non-zero): that is the expected return value in the documentation. This to me also implies that the other case on L283 also should be a `TEST_int_eq()` for consistency.